os/arch/arm/src/imxrt/imxrt_hyperflash.c

```diff
@@ -649,6 +849,68 @@ ssize_t up_progmem_write(size_t addr, const void *buf, size_t count)
 
     return count;
 }
+#else
+/************************************************************************************
+ * Name: up_flashinitialize
+ *
+ * Description:
+ *   Return an initialize MTD device instance.  MTD devices are not registered
+ *   in the file system, but are created as instances that can be bound to
+ *   other functions (such as a block or character driver front end).
+ *
+ ************************************************************************************/
+FAR struct mtd_dev_s *up_flashinitialize(void)
+{
+#ifdef CONFIG_MTD_PARTITION
+	/* Return the implementation-specific state structure as the MTD device */
+	int ret;
+	uint16_t p;
+	int partitionstomake;
+	uint32_t startblock;
+	uint32_t nblocks;
+	uint32_t blockspererase;
+	struct mtd_geometry_s geo;
+	struct mtd_dev_s *temp_mtd = NULL;
+#endif
+	if (flash_dev) {
+		goto err_out;
+	}
+	flash_dev = (FAR struct mtd_dev_s *)imxrt_mtd_initialize();
+	if (!flash_dev) {
+		goto err_out;
+	}
+
+#ifdef CONFIG_MTD_PARTITION
+	partitionstomake = sizeof(psize_info) / sizeof(psize_info[0]);
```

Comment:
The creation of MTD partitions is done based on the CONFIG_IMXRT_FLASH_PART_LIST config. 
The number of partitions, the name, type and size of each partition is determined by these configs.
Using a static array in code might not be the right way to do this. 

Also, the code in imxrt_bringup.c is already performing intitialization of the mtd partitions. 
So, this part of code might be a duplicate. 
Please re-check the entire flow for creating mtd partitions.